Framework/Algorithms/src/AppendSpectra.cpp

```diff
@@ -82,28 +84,28 @@ void AppendSpectra::exec() {
 
   const bool mergeLogs = getProperty("MergeLogs");
   const int number = getProperty("Number");
+  MatrixWorkspace_sptr output;
 
   if (event_ws1 && event_ws2) {
     // Both are event workspaces. Use the special method
-    MatrixWorkspace_sptr output = this->execEvent();
-    if (number > 1)
-      g_log.warning("Number property is ignored for event workspaces");
-    if (mergeLogs)
-      combineLogs(ws1->run(), ws2->run(), output->mutableRun());
-    // Set the output workspace
-    setProperty("OutputWorkspace", output);
-    return;
+    DataObjects::EventWorkspace_sptr e_output =
+        this->execEvent(*event_ws1, *event_ws2);
+    for (int i = 1; i < number; i++) {
+      e_output = this->execEvent(*e_output, *event_ws2);
+    }
+    output = boost::dynamic_pointer_cast<MatrixWorkspace>(e_output);
```

Comment:
Given that the result of the cast will be successful this may as well be a `static_pointer_cast`.